### PR TITLE
Update guide to zen2 and CrateDB 4.0 

### PR DESCRIPTION
**Pull request by mkleen**
Please do **NOT** merge this into master before CrateDB 4.0 is released.

### CODE REVIEW

File: docs/architecture/shared-nothing.txt [hunk 1]
```diff
@@ -125,39 +125,44 @@ cluster and coordinate operations:
...
+this could also be restricted to a subset of nodes if required.
+These nodes will then elect a single node as master to coordinate the cluster state
+across the cluster.
 
-To ensure that even in case of a network partition (when one part of the
-cluster is not available) the cluster requires a quorum (relative to the
-expected nodes in the cluster) to be able to elect the Metadata Primary.
+
+.. NOTE::
+   The following setting is only important for all 3 versions of CrateDB.
```
Comment (seut) [inline docs/architecture/shared-nothing.txt]: I don't understand this note, can you please elaborate what you want to express here?

File: docs/architecture/shared-nothing.txt [hunk 2]
```diff
@@ -99,8 +99,8 @@ operating on a shard level.
 Multi Node Setup: Clusters
 ==========================
 
-A CrateDB cluster is a set of two or more instances of CrateDB running on
-different hosts forming a single database unit. For inter-node communication
+A CrateDB cluster is a set of two or more instances of CrateDB referred as nodes
+running on different hosts forming a single database unit. For inter-node communication
```
Comment (andreidan) [inline docs/architecture/shared-nothing.txt]: This sentence is convoluted. I suggest:

"A CrateDB cluster is a set of two or more CrateDB instances, referred to as nodes, running on different hosts which form a single database unit"

File: docs/scaling/multi-node-setup.txt [hunk 3]
```diff
@@ -111,62 +120,62 @@ for more information.
 Master Node Election
 ====================
 
-Although all CrateDB nodes in a cluster are equal, one node is elected as
-master for managing cluster meta data. Like any other peer-to-peer system,
-nodes communicate with each other directly. The master node is responsible for
-making changes to, publishing the global cluster state, and delegating
-redistribution of shards when nodes join or leave the cluster. All nodes are
-eligible to become master.
+In a CrateDB cluster only one master node is responsible for making changes to the
```
Comment (andreidan) [inline docs/scaling/multi-node-setup.txt]: "In a CrateDB cluster only one master node " - this implies there are multiple master nodes.

===> "In a CrateDB cluster the master node is responsible ... "

File: docs/architecture/shared-nothing.txt [hunk 4]
```diff
@@ -125,39 +125,44 @@ cluster and coordinate operations:
...
-Metadata Primary to be elected. In a CrateDB cluster any node is eligible to be
-elected as a Metadata Primary, although this could also be restricted to a
-subset of nodes if required. There must be only one single Metadata Primary per
-cluster.
+The process of designating a single node as master node in a cluster is
+called Master Node Election. There must be only one single master node per cluster.
+In a CrateDB cluster any node is eligible to be elected as a master node, although
+this could also be restricted to a subset of nodes if required.
+These nodes will then elect a single node as master to coordinate the cluster state
+across the cluster.
```
Comment (andreidan) [inline docs/architecture/shared-nothing.txt]: I suggest we simplify this a bit: 

"The process of designating a single node as master node in a cluster is
called Master Node Election." 
=====> "The process of electing a node as the master node in a cluster is called Master Node Election"

"There must be only one single master node per cluster" ====> "There must be only one master node per cluster at any single time" (master nodes can change, the previous sentence implied a node is forever a master node)

File: docs/scaling/multi-node-setup.txt [hunk 5]
```diff
@@ -111,62 +120,62 @@ for more information.
...
-There may only be one master per cluster. To ensure this, CrateDB sets a quorum
-which needs to be present to elect a master and for the cluster to be
-operational.
+. NOTE::
+   The following setting only affects versions <= 4.x of CrateDB.
 
-This quorum can be configured using the ``minimum_master_nodes`` setting. It's
-highly recommend to set the quorum greater than half the maximum number of
-nodes in the cluster:
+To ensure this, CrateDB sets a quorum which needs to be present to elect a master and
```
Comment (andreidan) [inline docs/scaling/multi-node-setup.txt]: "To ensure this, CrateDB sets a quorum which needs to be present to elect a master and for the cluster to be operational" 
====> 
"To ensure this, CrateDB configures a quorum of nodes which need to be present to elect a master and participate in future master elections, keeping the cluster operational"

File: docs/scaling/multi-node-setup.txt [hunk 6]
```diff
@@ -111,62 +120,62 @@ for more information.
...
+for the cluster to be operational. This quorum can be configured using the ``minimum_master_nodes`` setting.
+It's highly recommend to set the quorum greater than half the maximum number of nodes in the cluster:
```
Comment (andreidan) [inline docs/scaling/multi-node-setup.txt]: "recommended"

**Comment by mkleen** [comment]
> with my changes squashed in

looks awesome, thank you!

**Pull request merged**